InnerEye/Common/generic_parsing.py

```diff
@@ -156,8 +172,38 @@ def list_or_dict(x: str) -> Union[Dict, List]:
 
             return p_type
 
+        def add_boolean_argument(parser: argparse.ArgumentParser, k: str, p: Parameter) -> None:
+            """
+            Add a boolean argument.
+            If the parameter default is False then allow --flag (to set it True) and --flag=Bool as usual.
+            If the parameter default is True then allow --no-flag (to set it to False) and --flag=Bool as usual.
+            :param parser: parser to add a boolean argument to.
+            :param k: argument name.
+            :param p: boolean parameter.
+            """
+            if not p.default:
+                # If the parameter default is False then use nargs="?" (argparse.OPTIONAL).
+                # This means that the argument is optional.
+                # If it is not supplied, i.e. in the --flag mode, use the "const" value, i.e. True.
+                # Otherwise, i.e. in the --flag=value mode, try to parse the argument as a bool.
+                parser.add_argument("--" + k, help=p.doc, type=parse_bool, default=False,
+                                    nargs=argparse.OPTIONAL, const=True)
+            else:
+                # If the parameter default is True then create an exclusive group of arguments.
+                # Either --flag=value as usual
+                # Or --no-flag to store False in the parameter k.
+                group = parser.add_mutually_exclusive_group(required=False)
+                group.add_argument("--" + k, help=p.doc, type=parse_bool)
+                group.add_argument('--no-' + k, dest=k, action='store_false')
+                parser.set_defaults(**{k: p.default})
```

Comment:
cool stuff - could you add documentation somewhere in .md about this and all the accepted bool values?

Reply:
Added a section to building_models.md. Also cleaned up the other docs when they use bool arguments